### PR TITLE
[DIA-3190] Remove `pubData` from `/messages`

### PR DESCRIPTION
**Pull request by bohdan-go-wombat**
JIRA - [DIA-3190](https://sourcepoint.atlassian.net/browse/DIA-3190)

[DIA-3190]: https://sourcepoint.atlassian.net/browse/DIA-3190?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by carmelo-iriti** [comment]
in this pr, we need just to remove the pubData query param in the HttpUrlManager class

**Comment by carmelo-iriti** [comment]
and reinforce the check in the related test with 
`queryParameter("pubData").assertNull()`

File: cmplibrary/src/main/java/com/sourcepoint/cmplibrary/campaign/CampaignManager.kt [hunk 1]
```diff
@@ -347,9 +347,6 @@ private class CampaignManagerImpl(
             accountId = spConfig.accountId.toLong(),
             authId = authId,
             propertyId = spConfig.propertyId.toLong(),
-            pubData = pubData?.toString()
```
Comment (carmelo-iriti) [inline cmplibrary/src/main/java/com/sourcepoint/cmplibrary/campaign/CampaignManager.kt:350]: this should stay

File: cmplibrary/src/main/java/com/sourcepoint/cmplibrary/campaign/CampaignManager.kt [hunk 2]
```diff
@@ -663,7 +660,6 @@ private class CampaignManagerImpl(
             ccpaMessageMetaData = null,
             gdprApplies = dataStorage.gdprApplies,
             ccpaApplies = dataStorage.ccpaApplies,
-            pubData = messageReq.pubData,
```
Comment (carmelo-iriti) [inline cmplibrary/src/main/java/com/sourcepoint/cmplibrary/campaign/CampaignManager.kt:666]: this should stay

File: cmplibrary/src/main/java/com/sourcepoint/cmplibrary/campaign/CampaignManager.kt [hunk 3]
```diff
@@ -677,7 +673,6 @@ private class CampaignManagerImpl(
             ccpaMessageMetaData = ccpaMessageMetaData,
             gdprApplies = dataStorage.gdprApplies,
             ccpaApplies = dataStorage.ccpaApplies,
-            pubData = messageReq.pubData,
```
Comment (carmelo-iriti) [inline cmplibrary/src/main/java/com/sourcepoint/cmplibrary/campaign/CampaignManager.kt:680]: this should stay

File: cmplibrary/src/main/java/com/sourcepoint/cmplibrary/data/network/model/optimized/MessagesApiModel.kt [hunk 4]
```diff
@@ -25,7 +25,6 @@ internal data class MessagesParamReq(
     @SerialName("metadataArg") val metadataArg: MetaDataArg?,
     @SerialName("body") val body: String,
     @SerialName("nonKeyedLocalState") val nonKeyedLocalState: JsonObject? = JsonObject(mapOf()),
-    @SerialName("pubData") val pubData: JsonObject = JsonObject(mapOf()),
```
Comment (carmelo-iriti) [inline cmplibrary/src/main/java/com/sourcepoint/cmplibrary/data/network/model/optimized/MessagesApiModel.kt:28]: this should stay

File: cmplibrary/src/main/java/com/sourcepoint/cmplibrary/data/network/util/HttpUrlManager.kt [hunk 5]
```diff
@@ -280,7 +280,6 @@ internal object HttpUrlManagerSingleton : HttpUrlManager {
             .addQueryParameter("env", param.env.queryParam)
             .addQueryParameter("nonKeyedLocalState", param.nonKeyedLocalState.toString())
             .addQueryParameter("localState", param.localState.toString())
-            .addQueryParameter("pubData", param.pubData.toString())
```
Comment (carmelo-iriti) [inline cmplibrary/src/main/java/com/sourcepoint/cmplibrary/data/network/util/HttpUrlManager.kt:283]: 👍🏻

**Review by carmelo-iriti — CHANGES_REQUESTED** [review]

File: cmplibrary/src/test/java/com/sourcepoint/cmplibrary/data/network/util/HttpUrlManagerTest.kt [hunk 6]
```diff
@@ -429,6 +429,7 @@ class HttpUrlManagerTest {
             queryParameter("nonKeyedLocalState").assertEquals("{}")
             queryParameter("metadata").assertEquals("""{  "ccpa": {    "applies": true  },  "gdpr": {    "applies": true  }}""")
             queryParameter("scriptVersion").assertEquals(BuildConfig.VERSION_NAME)
+            queryParameter("pubData").assertNull()
```
[the remark this answers is not in the document]
Comment (carmelo-iriti) [reply cmplibrary/src/test/java/com/sourcepoint/cmplibrary/data/network/util/HttpUrlManagerTest.kt:432]: this time we want to verify that those 2 fields are present in the url

**Review by Nevazhnovu — APPROVED** [review]

File: cmplibrary/src/test/java/com/sourcepoint/cmplibrary/data/network/util/HttpUrlManagerTest.kt (hunk 6 above)
[the remark this answers is not in the document]
Comment (carmelo-iriti) [reply cmplibrary/src/test/java/com/sourcepoint/cmplibrary/data/network/util/HttpUrlManagerTest.kt:432]: never mind , let me create a new story for it

**Review by carmelo-iriti — APPROVED** [review]

File: cmplibrary/src/test/java/com/sourcepoint/cmplibrary/data/network/util/HttpUrlManagerTest.kt (hunk 6 above)
[the remark this answers is not in the document]
Comment (bohdan-go-wombat) [reply cmplibrary/src/test/java/com/sourcepoint/cmplibrary/data/network/util/HttpUrlManagerTest.kt:432]: in this test we are not passing `localState` or `nonKeyedLocalState`, so I just asserted that it is `null` as well

[the remark this answers is not in the document]
Comment (bohdan-go-wombat) [reply cmplibrary/src/test/java/com/sourcepoint/cmplibrary/data/network/util/HttpUrlManagerTest.kt:432]: removed those changes, lets fix that in the dedicated story

**Pull request merged**